python/ray/serve/_private/deployment_scheduler.py

```diff
@@ -0,0 +1,259 @@
+import ray
+from typing import Callable, Dict, Tuple, Optional, List
+from dataclasses import dataclass
+from collections import defaultdict
+from ray._raylet import GcsClient
+from ray.serve._private.utils import get_all_node_ids
+from ray.util.scheduling_strategies import NodeAffinitySchedulingStrategy
+
+
+class SpreadDeploymentSchedulingPolicy:
+    pass
+
+
+class DriverDeploymentSchedulingPolicy:
+    pass
+
+
+@dataclass
+class DeploymentUpscaleRequest:
+    """Request to schedule a single replica.
```

Comment:
nit: naming is non-obvious (the name doesn't match the top-level description)

suggest `ReplicaSchedulingRequest` or similar